.github/scripts/should-run-job.sh

```
@@ -0,0 +1,135 @@
+
+# Get change categories
+CHANGES=$("$SCRIPT_DIR/detect-changes.sh")
+
+# Extract boolean values from JSON
+HAS_DOCS=$(echo "$CHANGES" | jq -r '.categories.has_docs // false')
+HAS_CODE=$(echo "$CHANGES" | jq -r '.categories.has_code // false')
+HAS_TESTS=$(echo "$CHANGES" | jq -r '.categories.has_tests // false')
+HAS_WORKFLOWS=$(echo "$CHANGES" | jq -r '.categories.has_workflows // false')
+HAS_CONFIG=$(echo "$CHANGES" | jq -r '.categories.has_config // false')
+DOCS_ONLY=$(echo "$CHANGES" | jq -r '.categories.docs_only // false')
+TESTS_ONLY=$(echo "$CHANGES" | jq -r '.categories.tests_only // false')
```

Shellcheck (0.11.0)

[warning] TESTS_ONLY appears unused. Verify use (or export if used externally).

(SC2034)